### DISCUSSION TITLE
Area in percentage of the image instead in pixels

### DISCUSSION THREAD

**Discussion by crisreimberg** [blakeblackshear/frigate#5180] [General]
Hi,
I am trying to reduce the notifications of false positives by creating a code that uses the main variables like area, ratio and score in a optimized way for my cameras, don't using the min_area, min_ratio, etc because I want to see the events in the Frigate to be able to optimize them. 

It said, I am having problems with the area variable that is sent to mqtt, because it is not the same value that is showed in snapshot of the event, could me explain if is there some average ou anything else that a have to calculate?

I am using it to convert this value to percentage to make it more easy to be configured, is that some reason to it be in pixels instead in %?

Thanks a lot, I am using frigate about 2 months and it is amazing, good job! :)

**Comment by NickM-27** [comment]
Area in pixels is a factor in percent based on the resolution, I'd say pixels is easier to use / think about but either way it works the same. 

The value showed in MQTT and the snapshot is the same just MQTT is for that event and the snapshot is when the snapshot was taken which can be different. They are calculated exactly the same.

**Reply by crisreimberg** [reply]
Thank you for your answer, I understood what was hapening with my calculations.
As a suggestion of improvement I give bellow some ideas to help in the debuging:

- Tool for view a Clip with detections ative, like in the Debug mode. I am doing it by setting a test camera with the clip link.
- An API option to get an update of an ative event on demand, it will make possible the development of custom algorithm tracks, for now I am thinking to create zones for quadrants of the image to get updates when a tracked object cross this quadrants.

Thanks.

**Reply by NickM-27** [reply]
> An API option to get an update of an ative event on demand, it will make possible the development of custom algorithm tracks, for now I am thinking to create zones for quadrants of the image to get updates when a tracked object cross this quadrants.

If I understand you correctly this is already supported in the api

**Reply by crisreimberg** [reply]
The improvement is to get an update on demand, because I just receive an update in the MQTT or api/events when the coditions in "should_update_db" function happens, right? I think in other situations the api just returns the last info saved and mqtt doesn't send an update.
